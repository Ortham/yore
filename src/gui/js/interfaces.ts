export interface Coordinates {
  latitude: number;
  longitude: number;
}

export interface Photo {
  path: string;
  location?: Location; // eslint-disable-line no-restricted-globals
  error?: string;
  height?: number;
  width?: number;
  loaded?: boolean;
}

interface Location {
  Existing?: Coordinates;
  Suggested?: [Coordinates, LocationAccuracy];
}

export interface LocationAccuracy {
  meters: number;
  seconds: number;
}

export interface LocationResponse {
  location?: Location; // eslint-disable-line no-restricted-globals
  error?: string;
}
