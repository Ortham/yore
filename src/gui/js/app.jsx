import 'babel-polyfill';
import 'whatwg-fetch';
import React from 'react';
import ReactDOM from 'react-dom';
import Page from './page';
import { getPhotos, getRootPath } from './requests';

function getInitialState() {
  const state = {
    rootPath: undefined,
    filterPhotos: false,
    currentPhoto: undefined,
    photos: []
  };

  return getRootPath()
    .then(responseBody => {
      state.rootPath = responseBody.rootPath;

      return getPhotos();
    })
    .then(photos => {
      state.photos = photos;

      return state;
    });
}

getInitialState().then(state => {
  ReactDOM.render(
    <Page rootPath={state.rootPath} photos={state.photos} />,
    document.getElementById('root')
  );
});
