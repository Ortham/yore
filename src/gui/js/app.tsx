import '@babel/polyfill';
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import 'whatwg-fetch';
import { Page, PageProps } from './page';
import {
  getInterpolate,
  getLocationHistoryPath,
  getPhotos,
  getRootPath,
  getLocations
} from './requests';
import { updatePhotoLocations } from './photo';

function getInitialState() {
  const state: PageProps = {
    interpolate: false,
    locationHistoryPath: undefined,
    photos: [],
    rootPath: undefined
  };

  return getRootPath()
    .then(responseBody => {
      state.rootPath = responseBody.rootPath || '';

      return getPhotos();
    })
    .then(photos =>
      getLocations(0, photos.length).then(locations =>
        updatePhotoLocations(photos, locations)
      )
    )
    .then(photos => {
      state.photos = photos;

      return getInterpolate();
    })
    .then(responseBody => {
      state.interpolate = responseBody.interpolate;

      return getLocationHistoryPath();
    })
    .then(responseBody => {
      state.locationHistoryPath = responseBody.locationHistoryPath || '';

      return state;
    });
}

getInitialState().then(state => {
  ReactDOM.render(<Page {...state} />, document.getElementById('root'));
});
