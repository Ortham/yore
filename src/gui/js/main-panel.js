import React from 'react';
import MapArea from './map-area';
import {locationDescription, hasSuggestion} from './photo';

export default function MainPanel(props) {
    if (props.photo) {
        return (
            <main>
                <section>
                    <img src={props.photo.src} />
                    <MapArea photo={props.photo} />
                </section>
                <footer>
                    {locationDescription(props.photo)}
                    <br />
                    <button
                        disabled={!hasSuggestion(props.photo)}
                        onClick={props.handleSuggestionApply} >
                        Apply
                    </button>
                    <button
                        disabled={!hasSuggestion(props.photo)}
                        onClick={props.handleSuggestionDiscard} >
                        Discard
                    </button>
                </footer>
            </main>
        );
    } else {
        return (
            <main>
                <section>No photo selected.</section>
                <footer>
                    <br />
                    <button disabled>Apply</button>
                    <button disabled>Discard</button>
                </footer>
            </main>
        );
    }
}
