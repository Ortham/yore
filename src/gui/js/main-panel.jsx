import React from 'react';
import PropTypes from 'prop-types';
import MapArea from './map-area';
import { locationDescription, hasSuggestion } from './photo';

export default function MainPanel(props) {
  if (props.photo) {
    return (
      <div id="main">
        <section>
          <img src={props.photo.src} alt="Selected" />
          <MapArea photo={props.photo} />
        </section>
        <footer>
          <div>{locationDescription(props.photo)}</div>
          <div>
            <button
              disabled={!hasSuggestion(props.photo)}
              onClick={props.handleSuggestionApply}
            >
              Apply
            </button>
            <button
              disabled={!hasSuggestion(props.photo)}
              onClick={props.handleSuggestionDiscard}
            >
              Discard
            </button>
          </div>
        </footer>
      </div>
    );
  }
  return (
    <div id="main">
      <section>No photo selected.</section>
      <footer>
        <div />
        <div>
          <button disabled>Apply</button>
          <button disabled>Discard</button>
        </div>
      </footer>
    </div>
  );
}

MainPanel.propTypes = {
  photo: PropTypes.shape({
    src: PropTypes.string
  }),
  handleSuggestionApply: PropTypes.func.isRequired,
  handleSuggestionDiscard: PropTypes.func.isRequired
};

MainPanel.defaultProps = {
  photo: undefined
};
