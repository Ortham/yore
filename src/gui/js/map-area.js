import React from 'react';
import GoogleMapReact from 'google-map-react';
import FaMapMarker from 'react-icons/lib/fa/map-marker';
import {googleMapsCoordinates} from './photo';

function MapMarker(props) {
    return <FaMapMarker lat={props.lat} lng={props.lng} style={{height: '24px', width: '24px', color: 'crimson'}} />;
}

export default function MapArea(props) {
    if (props.photo && props.photo.location) {
        const coord = googleMapsCoordinates(props.photo);
        return (
            <GoogleMapReact center={coord} zoom={5}>
                <MapMarker lat={coord.lat} lng={coord.lng} />
            </GoogleMapReact>
        );
    } else {
        return <GoogleMapReact center={[0, 0]} zoom={5} />;
    }
}
