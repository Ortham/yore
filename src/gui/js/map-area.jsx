import React from 'react';
import PropTypes from 'prop-types';
import GoogleMapReact from 'google-map-react';
import FaMapMarker from 'react-icons/lib/fa/map-marker';
import { googleMapsCoordinates } from './photo';

function MapMarker(props) {
  return (
    <FaMapMarker
      lat={props.lat}
      lng={props.lng}
      style={{
        height: '24px',
        width: '24px',
        color: 'crimson',
        position: 'relative',
        top: '-24px',
        left: '-12px'
      }}
    />
  );
}

MapMarker.propTypes = {
  lat: PropTypes.number.isRequired,
  lng: PropTypes.number.isRequired
};

export default function MapArea(props) {
  if (props.photo && props.photo.location) {
    const coord = googleMapsCoordinates(props.photo);
    return (
      <GoogleMapReact center={coord} zoom={5}>
        <MapMarker lat={coord.lat} lng={coord.lng} />
      </GoogleMapReact>
    );
  }
  return <GoogleMapReact center={[0, 0]} zoom={5} />;
}

MapArea.propTypes = {
  photo: PropTypes.shape({
    location: PropTypes.object
  })
};

MapArea.defaultProps = {
  photo: undefined
};
