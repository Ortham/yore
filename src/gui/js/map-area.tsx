import GoogleMapReact from 'google-map-react';
import * as React from 'react';
import { FaMapMarker } from 'react-icons/lib/fa';

import { Photo } from './interfaces';
import { googleMapsCoordinates } from './photo';

interface MapMarkerProps {
  lat: number;
  lng: number;
}

function MapMarker(props: MapMarkerProps) {
  return (
    <FaMapMarker
      {...props}
      style={{
        color: 'crimson',
        height: '24px',
        left: '-12px',
        position: 'relative',
        top: '-24px',
        width: '24px'
      }}
    />
  );
}

export interface MapAreaProps {
  photo?: Photo;
}

export function MapArea(props: MapAreaProps) {
  let coord = { lat: 0, lng: 0};
  let marker;
  if (props.photo && props.photo.location) {
    coord = googleMapsCoordinates(props.photo);
    marker = <MapMarker {...coord} />;
  }

  const style: React.CSSProperties = {
    height: '50%',
    position: 'relative',
    width: '100%'
  };

  return (
    <div style={style}>
      <GoogleMapReact center={coord} zoom={5}>
        {marker}
      </GoogleMapReact>
    </div>
  );
}
