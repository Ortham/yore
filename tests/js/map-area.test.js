import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('google-map-react', () => 'GoogleMapReact');
jest.mock('react-icons/lib/fa/map-marker', () => 'FaMapMarker');

import MapArea from '../../src/gui/js/map-area';

describe('MapArea', () => {
    test('renders a map centered at (0,0) if no photo prop is set', () => {
        const mapArea = renderer.create(<MapArea />).toJSON();
        expect(mapArea).toMatchSnapshot();
    });

    test('renders a map centered at (0,0) if the photo has no location', () => {
        let photo = {};
        const mapArea = renderer.create(<MapArea photo={photo} />).toJSON();
        expect(mapArea).toMatchSnapshot();
    });

    test('renders a map centered and with a marker at the photo\'s location if it has one', () => {
        let photo = {
            location: {
                Existing: {
                    latitude: 52.0,
                    longitude: 36.2
                }
            }
        };
        const mapArea = renderer.create(<MapArea photo={photo} />).toJSON();
        expect(mapArea).toMatchSnapshot();
    });
});
