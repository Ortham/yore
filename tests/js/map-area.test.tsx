import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Photo } from '../../src/gui/js/interfaces';

jest.mock('google-map-react', () => ({
  default: 'GoogleMapReact'
}));
jest.mock('react-icons/fi', () => ({
  FiMapPin: 'FiMapPin'
}));

import { MapArea } from '../../src/gui/js/map-area'; // eslint-disable-line import/first

describe('MapArea', () => {
  test('renders a map centered at (0,0) if no photo prop is set', () => {
    const mapArea = renderer.create(<MapArea />).toJSON();
    expect(mapArea).toMatchSnapshot();
  });

  test('renders a map centered at (0,0) if the photo has no location', () => {
    const photo: Photo = {
      path: ''
    };
    const mapArea = renderer.create(<MapArea photo={photo} />).toJSON();
    expect(mapArea).toMatchSnapshot();
  });

  test("renders a map centered and with a marker at the photo's location if it has one", () => {
    const photo = {
      location: {
        Existing: {
          latitude: 52.0,
          longitude: 36.2
        }
      },
      path: ''
    };
    const mapArea = renderer.create(<MapArea photo={photo} />).toJSON();
    expect(mapArea).toMatchSnapshot();
  });
});
