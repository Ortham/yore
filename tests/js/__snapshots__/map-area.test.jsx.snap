// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapArea renders a map centered and with a marker at the photo's location if it has one 1`] = `
<div
  style={
    Object {
      "position": "relative",
      "width": "50%",
    }
  }
>
  <GoogleMapReact
    center={
      Object {
        "lat": 52,
        "lng": 36.2,
      }
    }
    zoom={5}
  >
    <FaMapMarker
      lat={52}
      lng={36.2}
      style={
        Object {
          "color": "crimson",
          "height": "24px",
          "left": "-12px",
          "position": "relative",
          "top": "-24px",
          "width": "24px",
        }
      }
    />
  </GoogleMapReact>
</div>
`;

exports[`MapArea renders a map centered at (0,0) if no photo prop is set 1`] = `
<GoogleMapReact
  center={
    Array [
      0,
      0,
    ]
  }
  zoom={5}
/>
`;

exports[`MapArea renders a map centered at (0,0) if the photo has no location 1`] = `
<GoogleMapReact
  center={
    Array [
      0,
      0,
    ]
  }
  zoom={5}
/>
`;
